features/upgrade/storage/upgrade.feature

```diff
@@ -344,3 +344,120 @@ Feature: Storage upgrade tests
+      | ["metadata"]["name"]         | mypvc-ori |
+      | ["spec"]["storageClassName"] | gp2-csi   |
+
+    Given I obtain test data file "storage/misc/pod.yaml"
+    When I run oc create over "pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                         | mypod-ori    |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc-ori    |
+      | ["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/csipath |
+
+    Then the step should succeed
+    Given the pod named "mypod-ori" becomes ready
+
+    When I execute on the pod:
+      | touch   | /mnt/csipath/test-before-upgrade         |
+    Then the step should succeed
+    And I execute on the pod:
+      | sh | -c | sync -f /mnt/csipath/test-before-upgrade |
+
+    Given I ensure "mypod-ori" pod is deleted
+    
+    Given I obtain test data file "storage/csi/volumesnapshotclass-aws.yaml"
+    When I run oc create over "volumesnapshotclass-aws.yaml" replacing paths:
+      | ["metadata"]["name"] | csi-aws-ebs-snapshotclass |
+    Then the step should succeed
+   
+    Given I obtain test data file "storage/csi/volumesnapshot_v1.yaml"
+    When I run oc create over "volumesnapshot_v1.yaml" replacing paths:  
+      | ["metadata"]["name"]                            | csi-aws-ebs-snapshot      |
+      | ["spec"]["volumeSnapshotClassName"]             | csi-aws-ebs-snapshotclass |      
+      | ["spec"]["source"]["persistentVolumeClaimName"] | mypvc-ori                 |
+    Then the step should succeed
+    And I wait up to 200 seconds for the steps to pass:
+   
+    """
+    When I run the :describe client command with:
+      | resource | volumesnapshot       |
+      | name     | csi-aws-ebs-snapshot |
+    Then the output should match "Ready To Use\:\s+true"
+    """
+    
+    Given I obtain test data file "storage/csi/restorepvc.yaml"
+    Then I run oc create over "restorepvc.yaml" replacing paths:
+      | ["metadata"]["name"]           | mypvc-restore         |
+      | ["spec"]["storageClassName"]   | gp2-csi               |
+      | ["spec"]["dataSource"]["name"] |  csi-aws-ebs-snapshot |
+    Then the step should succeed
+
+    Given I obtain test data file "storage/misc/pod.yaml"
+    When I run oc create over "pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                         | mypod-restore |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc-restore |
+      | ["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/csipath  |
+    Given the pod named "mypod-restore" becomes ready
+
+    When I execute on the pod:
+      | ls | /mnt/csipath   |
+    Then the output should contain:
+      | test-before-upgrade |
+
+  # @author ropatil@redhat.com
+  @upgrade-check
+  @users=upuser1,upuser2
+  @admin
+  Scenario: Snapshot operator should work well before and after upgrade of cluster on AWS platform - check
```

Comment:
"- check" need to be removed.

Reply:
Thanks @duanwei33 , Changed the steps in upgrade-check as per your suggestions.